src/lightning_app/components/database/server.py

```diff
@@ -0,0 +1,159 @@
+                    self._client = None
+                    self.counter = 0
+
+                def run(self):
+                    self.db.run(token=self._private_token)
+
+                    if not self.db.alive():
+                        return
+
+                    if self.counter == 0:
+                        self._client = DatabaseClient(
+                            model=CounterModel,
+                            db_url=self.db.url,
+                            token=self._private_token,
+                        )
+
+                    rows = self._client.select_all()
+
+                    print(f"{self.counter}: {rows}")
+
+                    if not rows:
+                        self._client.insert(CounterModel(count=0))
+                    else:
+                        row: CounterModel = rows[0]
+                        row.count += 1
+                        self._client.update(row)
+
+                    if self.counter >= 100:
+                        row: CounterModel = rows[0]
+                        self._client.delete(row)
+                        self._exit()
+
+                    self.counter += 1
+
+            app = LightningApp(Flow())
+        """
+        super().__init__(parallel=True, cloud_build_config=BuildConfig(["sqlmodel"]))
+        self.db_filename = db_filename
+        self.debug = debug
+        self._models = models if isinstance(models, list) else [models]
+        self.drive = None
+
+    def run(self, token: Optional[str] = None) -> None:
+        """
+        Arguments:
+            token: Token used to protect the database access. Ensure you don't expose it through the App State.
+        """
+        self.drive = Drive("lit://database")
+        if self.drive.list(component_name=self.name):
+            self.drive.get(self.db_filename)
+            print("Retrieved the database from Drive.")
+
+        app = FastAPI()
+
+        create_database(self.db_filename, self._models, self.debug)
+        models = {m.__name__: m for m in self._models}
+        app.post("/select_all/")(SelectAll(models, token))
+        app.post("/insert/")(Insert(models, token))
+        app.post("/update/")(Update(models, token))
+        app.post("/delete/")(Delete(models, token))
+
+        sys.modules["uvicorn.main"].Server = DatabaseUvicornServer
+
+        run(app, host=self.host, port=self.port, log_level="error")
+
+    def alive(self) -> bool:
+        """Hack: Returns whether the server is alive."""
+        return self.db_url != ""
+
+    @property
+    def db_url(self) -> Optional[str]:
+        use_localhost = "LIGHTNING_APP_STATE_URL" not in os.environ
+        if use_localhost:
+            return self.url
+        if self.internal_ip != "":
+            return f"http://{self.internal_ip}:{self.port}"
+        return self.internal_ip
+
+    def on_exit(self):
+        self.drive.put(self.db_filename)
+        print("Stored the database to the Drive.")
```

Comment:
Is this the "new" way of persisting files? 
So far the designed approach was to use Path() on the attribute, and Lightning would persist these files automatically.

Reply:
Yes, Path would have been fine but I didn't implement the restore mechanism. So I decided to give it a go with Drive instead.